client/finality-grandpa/src/lib.rs

```diff
@@ -464,18 +464,18 @@ pub trait GenesisAuthoritySetProvider<Block: BlockT> {
 	fn get(&self) -> Result<AuthorityList, ClientError>;
 }
 
-impl<Block: BlockT, E> GenesisAuthoritySetProvider<Block>
-	for Arc<dyn ExecutorProvider<Block, Executor = E>>
+impl<Block: BlockT, E, Client> GenesisAuthoritySetProvider<Block> for Arc<Client>
```

Comment:
I had some doubts while reworking this.

Used in: https://github.com/paritytech/substrate/blob/911c9ab067d3cf487cfef61906ed2525cb87a5c9/bin/node/cli/src/service.rs#L193-L197,
https://github.com/paritytech/substrate/blob/911c9ab067d3cf487cfef61906ed2525cb87a5c9/client/finality-grandpa/src/lib.rs#L496-L501